packages/editor/src/components/post-locked-modal/index.js

```diff
@@ -0,0 +1,220 @@
+/**
+ * External dependencies
+ */
+import jQuery from 'jquery';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Modal, Button } from '@wordpress/components';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { addQueryArgs } from '@wordpress/url';
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { getWPAdminURL } from '../../utils/url';
+import PostPreviewButton from '../post-preview-button';
+
+class PostLockedModal extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.sendPostLock = this.sendPostLock.bind( this );
+		this.receivePostLock = this.receivePostLock.bind( this );
+		this.releasePostLock = this.releasePostLock.bind( this );
+	}
+
+	componentDidMount() {
+		jQuery( document )
+			.on( 'heartbeat-send.refresh-lock', this.sendPostLock )
+			.on( 'heartbeat-tick.refresh-lock', this.receivePostLock )
+			.on( 'beforeunload.edit-post', this.releasePostLock );
+	}
```

Comment:
If there's a good reference to these three events, it could be good to add URLs here.